Release notes for v4.0.0-preview3 (#1363)
dotnet/SqlClient: CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

## [Preview Release 4.0.0-preview3.x.x] - 2021-10-20

This update brings the below changes over the previous release:

### Breaking changes over preview release v4.0.0-preview2

- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)

> - Exception will not be thrown if a User ID is provided in the connection string when using `ActiveDirectoryIntegrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)

Reviewer: Could you mention this would be a breaking change on net fx?

### Added

- Add `GetFieldValue(Async)<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019)

### Fixed

> - Fixed `FormatException` when opening a connection with event tracing enabled [#1290](https://github.com/dotnet/SqlClient/pull/1290)

Reviewer: ```suggestion
- Fixed `FormatException` when opening a connection with event tracing enabled [#1291](https://github.com/dotnet/SqlClient/pull/1291)
```

- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1327](https://github.com/dotnet/SqlClient/pull/1327)
- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1335](https://github.com/dotnet/SqlClient/pull/1335)

> - Fixed data corruption issues by reverting changes to async cancellations [#1344](https://github.com/dotnet/SqlClient/pull/1344)

Reviewer: ```suggestion
- Fixed data corruption issues by reverting changes to async cancellations [#1352](https://github.com/dotnet/SqlClient/pull/1352)
```

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)

## [Preview Release 4.0.0-preview3.x.x] - 2021-10-20

This update brings the below changes over the previous release:

### Breaking changes over preview release v4.0.0-preview2

- Dropped support for .NET Core 2.1 [#1272](https://github.com/dotnet/SqlClient/pull/1272)

> - [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `ActiveDirectoryIntegrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)

Reviewer 2: ```suggestion
- [.NET Framework] Exception will not be thrown if a User ID is provided in the connection string when using `Active Directory Integrated` authentication [#1359](https://github.com/dotnet/SqlClient/pull/1359)
```

### Added

- Add `GetFieldValueAsync<T>` and `GetFieldValue<T>` support for `XmlReader`, `TextReader`, `Stream` [#1019](https://github.com/dotnet/SqlClient/pull/1019)

### Fixed

- Fixed `FormatException` when opening a connection with event tracing enabled [#1291](https://github.com/dotnet/SqlClient/pull/1291)

> - Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1327](https://github.com/dotnet/SqlClient/pull/1327)

Reviewer: ```suggestion
- Fixed improper initialization of `ActiveDirectoryAuthenticationProvider` [#1328](https://github.com/dotnet/SqlClient/pull/1328)
```

> - Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1335](https://github.com/dotnet/SqlClient/pull/1335)

Reviewer: ```suggestion
- Fixed `MissingMethodException` when accessing `SqlAuthenticationParameters.ConnectionTimeout` [#1336](https://github.com/dotnet/SqlClient/pull/13365)
```

- Fixed data corruption issues by reverting changes to async cancellations [#1352](https://github.com/dotnet/SqlClient/pull/1352)
- Fixed performance degradation by reverting changes to MARS state machine [#1357](https://github.com/dotnet/SqlClient/pull/1357)

> - Fixed bug where environment variables are ignored when using `ActiveDirectoryDefault` authentication [#1360](https://github.com/dotnet/SqlClient/pull/1360)

Reviewer 2: ```suggestion
- Fixed bug where environment variables are ignored when using `Active Directory Default` authentication [#1360](https://github.com/dotnet/SqlClient/pull/1360)
```
